src/testdrive/src/util/kinesis.rs

```diff
@@ -10,50 +10,57 @@
 use std::convert::TryFrom;
 use std::time::Duration;
 
-use rusoto_kinesis::{DescribeStreamInput, Kinesis, KinesisClient};
+use crate::action::State;
+use rusoto_kinesis::{DescribeStreamInput, Kinesis};
 
 use ore::retry;
 
-pub const DEFAULT_KINESIS_TIMEOUT: Duration = Duration::from_millis(12700);
-
 pub async fn wait_for_stream_shards(
-    kinesis_client: &KinesisClient,
+    state: &State,
```

Comment:
In terms of module boundaries, these utility functions are meant to be fairly general purpose and so should not depend on `State`! Would be best to just pass in an explicit `timeout` argument here, and let the caller figure out what timeout to use.